SproxydKeysScan/DuplicateKeysIngestion.js

```diff
@@ -0,0 +1,279 @@
+     * @property {number} info.cseq - total number of logged objects in this raft session.
+     * @property {number} info.prune - ???
+       @property {Array.<RaftJournalLogObject>} log - Array of RaftJournalLogObject
+     */
+
+    /**
+     * Fetches RaftJournalBody
+     * @param {function(err: Error, res: RaftJournalBody)} cb - callback
+     * @returns {undefined}
+     * @callback called with (err:Error, body:RaftJournalBody)
+     */
+    getBatch(cb) {
+        const requestUrl = `${this.url}/log?begin=${this.begin}&limit=${this.limit}&targetLeader=False`;
+        return this._httpRequest('GET', requestUrl, null, (err, res) => {
+            if (err) {
+                return cb(err);
+            }
+            if (!res || !res.body) {
+                return cb(new Error(`GET ${requestUrl} returned empty body`));
+            }
+
+            if (res.statusCode !== 200) {
+                return cb(new Error(`GET ${requestUrl} returned status ${res.statusCode}`));
+            }
+
+            const body = JSON.parse(res.body);
+            return cb(null, body);
+        });
+    }
+    /**
+     * @typedef {Object} ExtractedKey
+     * @property {string} ObjectKey - unversioned object key
+     * @property {Array.<string>} sproxydKeys - Array of sproxydKeys for the given object key
+     */
+
+    /**
+     * Extracts Sproxyd Keys for a given object key
+     * @param {RaftJournalBody} body - RaftJournalBody instance
+     * @param {function(err: Error, res: Array.<ExtractedKey>)} cb - callback
+     * @returns {undefined}
+     * @callback called with (err:Error, extractedKeys:Array.<ExtractedKey>)
+     */
+    processBatch(body, cb) {
+        this._setCseq(body);
+
+        // { objectKey, sproxydKeys }
+        const extractedKeys = [];
+
+        body.log.forEach(record => {
+            if (record.method === 8) {
+                record.entries.forEach(entry => {
+                    if (!entry.value) {
+                        return;
+                    }
+                    const objectKey = entry.key;
+                    if (!objectKey.includes('\u0000')) {
+                        // skip non-versioned objectKeys
+                        return;
+                    }
+
+                    const bucket = record.db;
+
+                    let json = null;
+                    try {
+                        json = JSON.parse(entry.value);
+                    } catch (err) {
+                        log.error('json corrupted');
+                        return;
+                    }
+
+                    // skip empty objects
+                    if (!json.location || !Array.isArray(json.location)) {
+                        return;
+                    }
+                    const sproxydKeys = json.location.map(loc => loc.key);
+                    extractedKeys.push({ objectKey, sproxydKeys, bucket });
+                });
+            }
+        });
+        log.info('processBatch succeeded');
```

Comment:
```suggestion
        log.debug('processBatch succeeded', {
            begin: this.begin,
            limit: this.limit,
        });
```
This can be useful as debug information, but may be too low-level for info, ideally, in info level we log a regular status update (`logProgress`) with:
- how many total records and entries processed in that time period
- what is the current offset
- the latest cseq
- the lag (as `cseq - next_begin`)
- and also maybe the time spent processing the log (compared to idling) may be useful to diagnose potential load issues